soccer/src/soccer/strategy/agent/position/defense.cpp

```diff
@@ -5,34 +5,16 @@ namespace strategy {
 Defense::Defense(int r_id) : Position(r_id) { position_name_ = "Defense"; }
 
 std::optional<RobotIntent> Defense::derived_get_task(RobotIntent intent) {
-    if (check_is_done()) {
-        // toggle move pts
-        move_ct_++;
-    }
 
     // thread-safe getter
     WorldState* world_state = this->world_state();
 
-    // oscillate along horizontal line (temp)
-    double x = -3.0;
-    if (move_ct_ % 2 == 1) {
-        x = 3.0;
-    }
-    rj_geometry::Point target_pt{x, 3.0};
+    // Get Ball Location
+    rj_geometry::Point ball_location{world_state->ball.position};
 
-    // stop at end of path
-    rj_geometry::Point target_vel{0.0, 0.0};
-
-    // face ball
-    planning::PathTargetFaceOption face_option = planning::FaceBall{};
-
-    // avoid ball
-    bool ignore_ball = false;
-
-    planning::LinearMotionInstant goal{target_pt, target_vel};
-    intent.motion_command = planning::PathTargetMotionCommand{goal, face_option, ignore_ball};
-    intent.motion_command_name = "path_target";
-    return intent;
+    // Create Waller
+    Waller* waller = new Waller();
```

Comment:
The check warning here is not that helpful. Basically it's confused because you don't need to create pointers to objects to use them in c++. In Java the new keyword is the only way to create a new object, but in C++ you can initialize something like this:
```
Waller waller{};
```
which you've done for some of the rj_geometry::Point types below already. Then you just have to change the call to get_task below to be `waller.get_task()`

https://www.learncpp.com/cpp-tutorial/classes-and-class-members/